artemis/modules/api_scanner.py

```diff
@@ -0,0 +1,168 @@
+import json
+import os
+import tempfile
+from typing import Any, Dict, Optional
+
+from karton.core import Task
+from openapi_schema_validator import validate
+from openapi_spec_validator.readers import read_from_filename
+from pydantic import BaseModel
+
+from artemis import http_requests, load_risk_class
+from artemis.binds import Service, TaskStatus, TaskType
+from artemis.config import Config
+from artemis.module_base import ArtemisBase
+from artemis.task_utils import get_target_url
+
+COMMON_SPEC_PATHS = [
+    "/swagger.json",
+    "/v2/swagger.json",
+    "/v3/api-docs",
+    "/openapi.json",
+    "/api-docs",
+    "/api/docs" "/docs/swagger.json",
+]
+
+
+class APIResult(BaseModel):
+    url: str
+    method: str
+    vulnerable: bool
+    details: Optional[str]
+    curl_command: Optional[str]
+    status_code: Optional[int]
+
+
+@load_risk_class.load_risk_class(load_risk_class.LoadRiskClass.MEDIUM)
+class APIScanner(ArtemisBase):
+    """
+    Scans API endpoints for vulnerabilities based on OpenAPI/Swagger specifications.
+    """
+
+    identity = "api_scanner"
+    filters = [
+        {"type": TaskType.SERVICE.value, "service": Service.HTTP.value},
+    ]
+
+    def discover_spec(self, base_url: str) -> Optional[str]:
+        """Try to discover OpenAPI/Swagger specification from common paths."""
+        for path in COMMON_SPEC_PATHS:
+            try_url = base_url.rstrip("/") + path
+            try:
+                response = http_requests.get(try_url)
+                if response.status_code == 200 and (
+                    "openapi" in response.text.lower() or "swagger" in response.text.lower()
+                ):
+                    with tempfile.NamedTemporaryFile(delete=False, suffix=".json") as f:
+                        f.write(response.content)
+                        temp_file = f.name
+
+                    # Validate the spec
+                    spec_dict, _ = read_from_filename(temp_file)
+                    try:
+                        validate(spec_dict)
+                        return temp_file
+                    except Exception as e:
+                        self.log.info(f"Unable to validate spec at {try_url}: {e}")
+                        continue
+            except Exception as e:
+                self.log.debug(f"Error checking {try_url}: {e}")
+                continue
+        return None
+
+    def install_offat(self) -> None:
+        if not os.path.exists("/offat"):
+            os.system("git clone https://github.com/OWASP/OFFAT /offat")
```

Comment:
I'd rather use subprocess.call, it's less prone to command injection if we start building these strings somehow